core/command_line_parser.h

```diff
@@ -0,0 +1,246 @@
+	Map<const CommandLineOption *, PackedStringArray> _parsed_prefixes;
+	Map<const CommandLineOption *, int> _parsed_count;
+
+	String _help_header;
+	String _help_footer;
+	String _usage_title;
+
+	String _error;
+
+	float _min_option_simmilarity = 0.3;
+
+	int _left_help_pad = 2;
+	int _right_help_pad = 4;
+	int _help_line_length = 80;
+	int _min_description_length = _help_line_length / 2;
+
+	bool _autogenerate_usage = true;
+	bool _allow_forwarding_args = false;
+	bool _allow_adjacent = true;
+	bool _allow_sticky = true;
+	bool _allow_compound = true;
+
+	// Parser main helpers
+	bool _is_options_valid() const;
+	void _read_default_args();
+	int _validate_arguments(int p_current_idx); // Returns number of arguments taken, -1 on validation error.
+
+	// Helpers for the function above that parse a specific case
+	int _validate_positional(const String &p_arg, int p_current_idx);
+	int _validate_adjacent(const String &p_arg, const String &p_prefix, int p_separator);
+	int _validate_short(const String &p_arg, const String &p_prefix, int p_current_idx);
+	int _validate_long(const String &p_arg, const String &p_prefix, int p_current_idx);
+
+	// Validation helpers
+	const CommandLineOption *_validate_option(const String &p_name, const String &p_prefix);
+	int _validate_option_args(const CommandLineOption *p_option, const String &p_display_name, int p_current_idx, bool p_skip_first = false);
+	bool _validate_option_arg(const CommandLineOption *p_option, const String &p_display_name, const String &p_arg);
+
+	// Save information about parsed option
+	void _save_parsed_option(const CommandLineOption *p_option, const String &p_prefix, int p_idx, int p_args_count, const String &p_additional_value = String());
+	void _save_parsed_option(const CommandLineOption *p_option, const String &p_prefix, const String &p_value = String());
+	void _save_parsed_option(const CommandLineOption *p_option, int p_idx, int p_args_count);
+
+	// Help text printers
+	String _get_usage(const Vector<Pair<const CommandLineOption *, String>> &p_printable_options) const;
+	String _get_options_description(const OrderedHashMap<String, PackedStringArray> &p_categories_data) const;
+
+	// Other utilies
+	String _to_string(const PackedStringArray &p_names) const; // Returns all option names separated by commas with all prefix variants.
+	String _get_prefixed_longest_name(const PackedStringArray &p_names) const; // Returns longest name with first available prefix (long or short).
+	ParsedPrefix _parse_prefix(const String &p_arg) const;
+	String _find_most_similar(const String &p_name) const;
+	static bool _contains_optional_options(const Vector<Pair<const CommandLineOption *, String>> &p_printable_options);
+
+protected:
+	static void _bind_methods();
+
+public:
+	Error parse_args(const PackedStringArray &p_args);
+	Error validate();
+
+	void add_option(const Ref<CommandLineOption> &p_option);
+	int get_options_count() const;
```

Comment:
```suggestion
	int get_option_count() const;
```

See methods like `get_button_count()`, `get_child_count()` throughout Godot's API.